manifest/v1alpha/twindow/twindow.go

```diff
@@ -112,21 +112,27 @@ type monthPeriodCalculator struct{}
 
 func (monthPeriodCalculator) periodsCountDiff(timeWindow CalendarTimeWindow, date Date) int {
 	timeWindowYear, timeWindowMonth, timeWindowDay := timeWindow.DateWithTime.Date()
-	timestampYear, timestampMonth, timestampDay := date.Date()
+	timestampYear, timestampMonth, _ := date.Date()
 
 	// This corrects for the situation when starting date of monthly calendar time window is at a day that doesn't
 	// occur in certain months. For example, if the starting date is 31st of January and the timestamp is 1st of March
 	// then the difference in months should be 1 and not 2. This mirrors the way AddDate normalizes dates, where
 	// 31st of February is normalized to 3rd of March.
 	var correction int
-	daysInPreviousMonth := daysIn(timestampMonth-1, timestampYear)
-	if timeWindowDay > daysInPreviousMonth && timestampDay <= timeWindowDay-daysInPreviousMonth {
+	if isDateNormalized(timeWindowDay, date) {
 		correction = -1
 	}
 
 	return (timestampYear-timeWindowYear)*int(monthsInYear) + int(timestampMonth) - int(timeWindowMonth) + correction
 }
 
+func isDateNormalized(timeWindowStartDay int, date Date) bool {
+	timestampYear, timestampMonth, timestampDay := date.Date()
+	daysInPreviousMonth := daysIn(timestampMonth-1, timestampYear)
+	wasDateNormalized := timeWindowStartDay > daysInPreviousMonth && timestampDay <= timeWindowStartDay-daysInPreviousMonth
+	return wasDateNormalized
+}
```

Comment:
Maybe we can simplify this:

```suggestion
func normalizeDate(timeWindowStartDay int, date Date) int {
	timestampYear, timestampMonth, timestampDay := date.Date()
	daysInPreviousMonth := daysIn(timestampMonth-1, timestampYear)
	isNormalized := timeWindowStartDay > daysInPreviousMonth && timestampDay <= timeWindowStartDay-daysInPreviousMonth
  if isNormalized {
    return -1
  }
  return 0
}
```

then in both your use cases you could just do:

```go
correction := normalizeDate(timeWindowDay, date)
```

Reply:
Ok, check now.